src/cursor/abstract_cursor.ts

```diff
@@ -903,7 +866,7 @@ class ReadableCursorStream extends Readable {
   }
 
   override _destroy(error: Error | null, callback: (error?: Error | null) => void): void {
-    this._cursor.close(err => process.nextTick(callback, err || error));
+    this._cursor.close().finally(() => callback(error));
```

Comment:
I think close can still error, no? in that case, we can still callback here with that error instead of ignoring it `.close().then(() => callback(error), err => callback(err))`, or is that not a concern?

Reply:
close should never throw so this should be fine, inside the cursor.close method the callback is never invoked with an error

Reply:
If we're going to rely on the assumption that .close() never rejects, we should enforce that via testing.  The problem is that we cannot.  How do you write a test that says "this method never throws"?  What conditions are you setting up in that test to make it fail if it does throw at some point in the future?  It's impossible.  

Using callbacks we could enforce it by simply making the callback `(value) => { .. } ` and not taking an error.  Using promises it is not possible.

I agree with Daria.  Because we can't encode this assumption into tests to enforce it for the future, we can't rely on it here and we should handle the error case.

Reply:
fixed